week09/time-needed-to-inform-all-employees.cpp

```diff
@@ -0,0 +1,29 @@
+// TC: O(N + E), where N and E = total number of nodes and edges
+// MC: O(H), where H = height of tree
+class Solution {
+public:
+	vector<vector<int>>adjList;
+
+	int findMinTime(int root, vector<int> &informTime){
+		int maxTime = 0;
+
+		for(int i : adjList[root]){
+			maxTime = max(maxTime, findMinTime(i, informTime));
+		}
+
+		return maxTime + informTime[root];
+	}
+
+	int numOfMinutes(int n, int headID, vector<int>& manager, vector<int>& informTime) {
+        adjList.clear();
+		adjList.resize(n);
```

Comment:
This is actually very risky! Resize does not CLEAR the old data. So if you do the following operations in order, you will get error:

```c++
Solution sol = new Solution();
cout << sol.numOfMinutes(n1, head1, manager1, informTime1) << "\n";
cout << sol.numOfMinutes(n2, head2, manager2, informTime2) << "\n"; // sol will contain the older data from the first operation
```

So, clear the adjList always.

Reply:
vaia its already cleared at 18th line